odoo/addons/base/models/res_partner.py

```diff
@@ -793,7 +793,7 @@ def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_ui
                                vat=unaccent('res_partner.vat'),)
 
             where_clause_params += [search_name]*3  # for email / display_name, reference
-            where_clause_params += [re.sub('[^a-zA-Z0-9]+', '', search_name) or None]  # for vat
+            where_clause_params += [re.sub('[^a-zA-Z0-9-\-]+', '', search_name) or None]  # for vat
```

Comment:
```suggestion
            where_clause_params += [re.sub('[^a-zA-Z0-9\-\.]+', '', search_name) or None]  # for vat
```

`-` is used to defined a range of values. The one before `\-` should be removed.
I think you can also keep `.` as some countries are using it (i.e. `ch` and `co`)
You can check the list here: https://github.com/odoo/odoo/blob/54b6a7a97e574249c22f16ea2756989e4bd4fac3/addons/base_vat/models/res_partner.py#L25

Reply:
I agree with your explanation. Thanks for review!